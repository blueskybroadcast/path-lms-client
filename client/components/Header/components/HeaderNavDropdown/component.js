import React from 'react';
import classNames from 'classnames';

class HeaderNavDropdown extends React.Component {
  state = {
    isOpen: false
  }

  handleDropdownOpen = () => {
    this.setState((prevState) => ({
      isOpen: !prevState.isOpen
    }));
  }

  render() {
    return (
      <div
        className={classNames('item dropdown', this.state.isOpen && 'open')}
        onMouseEnter={this.handleDropdownOpen}
        onMouseLeave={this.handleDropdownOpen}
      >
        <a
          id="events-drop"
          role="button"
          className="dropdown-toggle"
          title="Events"
          href="/showroom/events"
        >
          <i className="icon icon-event" />
          &nbsp;Events&nbsp;
          <b className="caret" />
        </a>
        <ul className="dropdown-menu" role="menu">
          <li role="presentation">
            <a
              id="live-events"
              role="menuitem"
              tabIndex="-1"
              href="/showroom/events#live-events-content"
            >
              Live Events
            </a>
          </li>
          <li role="presentation">
            <a
              id="on-demand-events"
              role="menuitem"
              tabIndex="-1"
              href="/showroom/events#on-demand-events-content"
            >
              On-Demand Events
            </a>
          </li>
        </ul>
      </div>
    );
  }
}

export default HeaderNavDropdown;
