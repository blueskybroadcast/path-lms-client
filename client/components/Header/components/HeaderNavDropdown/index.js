import HeaderNavDropdown from './component';

export default HeaderNavDropdown;
