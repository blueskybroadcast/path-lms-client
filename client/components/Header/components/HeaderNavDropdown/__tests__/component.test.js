import React from 'react';
import { shallow } from 'enzyme';
import HeaderNavDropdown from '../component';

describe('HeaderNavDropdown', () => {
  test('should render HeaderNavDropdown correctly', () => {
    const wrapper = shallow(
      <HeaderNavDropdown />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
