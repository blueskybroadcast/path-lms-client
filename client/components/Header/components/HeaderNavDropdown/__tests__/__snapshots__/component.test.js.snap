// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNavDropdown should render HeaderNavDropdown correctly 1`] = `
<div
  className="item dropdown"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    className="dropdown-toggle"
    href="/showroom/events"
    id="events-drop"
    role="button"
    title="Events"
  >
    <i
      className="icon icon-event"
    />
     Events 
    <b
      className="caret"
    />
  </a>
  <ul
    className="dropdown-menu"
    role="menu"
  >
    <li
      role="presentation"
    >
      <a
        href="/showroom/events#live-events-content"
        id="live-events"
        role="menuitem"
        tabIndex="-1"
      >
        Live Events
      </a>
    </li>
    <li
      role="presentation"
    >
      <a
        href="/showroom/events#on-demand-events-content"
        id="on-demand-events"
        role="menuitem"
        tabIndex="-1"
      >
        On-Demand Events
      </a>
    </li>
  </ul>
</div>
`;
